modules/frontend/querysharding.go

```diff
@@ -190,12 +215,18 @@ func mergeResponses(ctx context.Context, rrs []RequestResponse) (*http.Response,
 	}
 
 	if errCode == http.StatusOK {
+		statusCode := http.StatusOK
+		if totalBlockErrCount > 0 {
+			// If there are failed blocks, and we haven't returned with an error,
+			// signal the upstream that the result contains partial results.
+			statusCode = http.StatusPartialContent
+		}
 		return &http.Response{
-			StatusCode: http.StatusOK,
-			Body:       ioutil.NopCloser(bytes.NewReader(combinedTrace)),
+			StatusCode: statusCode,
+			Body:       ioutil.NopCloser(bytes.NewReader(combinedTraceBytes)),
```

Comment:
```suggestion
			Body:       io.NopCloser(bytes.NewReader(combinedTraceBytes)),
```
`ioutil.NopCloser` has been moved to `io.NopCloser`, see https://golang.org/doc/go1.16#ioutil
We just removed all occurrences of `ioutil` here #998 🙂